accepted/future-releases/generalized-typedef-2018/feature-specification.md

```diff
@@ -114,32 +119,34 @@ evaluated as an expression, it is subject to instantiation to bound.
 *This treatment of generic type aliases is again the same as it was
 previously, but it involves a larger set of types.*
 
-A type alias application of the form _F_ or the form
+A type alias application _A_ of the form _F_ or the form
 _F&lt;T<sub>1</sub>..T<sub>k</sub>&gt;_ can be used as a type annotation,
 as a type argument, as part of a function type or a function signature, as
 a type literal, in an `on` clause of a `try` statement, in a type test
-(`e is F`), and in a type cast (`e as F`).
-
-**The following is under discussion. Exactly one rule will be confirmed.
-The following two variants are considered, along with small variations or
-hybrids thereof:**
-
-*  A type alias can _not_ be used in a position where a class is
-   expected: in the `extends`, `with`, `implements`, or `on` clause of a
-   class or mixin declaration; for a static member access; or in an
-   instance creation expression (`new F()`, `const F<int>()`). OR
-
-*  A type alias can _not_ be used for static member accesses. When a type
-   alias _F_ resp. _F&lt;T<sub>1</sub>..T<sub>k</sub>&gt;_ occurs as the
-   entity that a class `extends` and when it occurs in a `with` clause or
-   `implements` clause of a class or an `on` clause of a mixin, or it is
-   used in an instance creation expression, it is a compile-time error
-   unless it denotes a class, otherwise it is treated as if that class had
-   been specified explicitly.
-   (*E.g., `class C extends myPrefix.F<int> {}` would be equivalent to
-   `class C extends D<List<int>, int> {}` if the library imported as
-   `myPrefix` contains `typedef F<X> = D<List<X>, X>;`, assuming that
-   `D` is accessible to the current library and has no prefix.*)
+(`e is A`), and in a type cast (`e as A`).
+
+A type alias application of the form _F_ or the form
+_F&lt;T<sub>1</sub>..T<sub>k</sub>&gt;_ denoting a class can be used in a
+position where a class is expected: in the `extends`, `with`, `implements`,
+and `on` clause of a class or mixin declaration; and in an instance
+creation expression (*e.g., `new F()`, `const F<String>.named('Hello!')`*).
+
+A type alias application _A_ of the form _F_ or the form
```

Comment:
@lrhn Do we intend to support the actual syntax `F<T>.staticMethod` now (as opposed to just implicitly via `G.staticMethod` where `G` is a type alias for `F<T>`?  I don't particularly object, but if so we should call out that the non-typedef version is allowed now as well, since this is a change.  I presume there are no parsing issues, since named parameters essentially cover this corner of the grammar already?

Reply:
I would say no — for now.

I do want to support `something<typearg>` as a syntax in more places (`List<int>` as a type literal, `object.genericMethod<int>` as an instantiated tear-off, likely other), but I don't see an advantage in adding piecemeal syntax support for it until we do the full thing.

There is no current *benefit* to allowing the syntax `F<T>.foo` for static access (we currently only allow it only for constructor invocations); you can always do `F.foo` and get the same thing, so this would require be a parser change for no actual benefit.

Reply:
Sounds good! ;-)

I uploaded a new version where it is an error to invoke a static method using syntax like `F<int>.m(42)` where `F` is a type alias.